pkg/pillar/cmd/nodeagent/handlebaseos.go

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) 2017-2018 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+// base os event handlers
+
+package nodeagent
+
+import (
+	"fmt"
+	"github.com/lf-edge/eve/pkg/pillar/cast"
+	"github.com/lf-edge/eve/pkg/pillar/types"
+	log "github.com/sirupsen/logrus"
+)
+
+func lookupBaseOsStatusOnPartition(ctx *nodeagentContext, key string) *types.BaseOsStatus {
+	sub := ctx.subBaseOsStatus
+	items := sub.GetAll()
+	for _, st := range items {
+		status := cast.CastBaseOsStatus(st)
+		if status.PartitionLabel == key {
+			return &status
+		}
+	}
+	return nil
+}
+
+// mark the zedcloud health/connectivity test complete flag
+// for baseosmgr to pick up and complete the partition activation
+func initiateBaseOsZedCloudTestComplete(ctx *nodeagentContext) {
+	if !ctx.updateInprogress {
+		return
+	}
+	log.Infof("initiateBaseOsZedCloudTestComplete(%s)\n", ctx.curPart)
+	// get the current partition zboot config and status
+	zbootConfig := lookupZbootConfig(ctx, ctx.curPart)
+	zbootStatus := lookupZbootStatus(ctx, ctx.curPart)
+	if zbootStatus == nil || zbootConfig == nil {
+		log.Errorf("zboot(%s) status/config get fail\n", ctx.curPart)
+		return
+	}
+	// get the baseos status information, for current partition
+	status := lookupBaseOsStatusOnPartition(ctx, ctx.curPart)
+	if status == nil {
+		log.Errorf("baseOs(%s) status get fail\n", ctx.curPart)
+		return
+	}
+	// the installed and configured version do not match
+	if status.BaseOsVersion != zbootStatus.ShortVersion {
```

Comment:
This seems like an opportunity for a device to become un-updatable, because when we don't signal TestComplete to baseosmgr to clear the inprogress bit, I don't think we can update to another baseos image. At best we can recover by rebooting.

So ... why do you need to do this check here?

Reply:
removed the check.